src/pointer.rs

```diff
@@ -315,39 +324,33 @@ impl Pointer {
     }
 
     /// Finds the commonality between this and another `Pointer`.
+    #[must_use]
     pub fn intersection<'a>(&'a self, other: &Self) -> &'a Self {
-        if &self.0 == "/" || &other.0 == "/" {
-            if self.0 != other.0 {
-                return Self::root();
-            } else {
-                return self;
-            }
+        if self.is_root() || other.is_root() {
+            return Self::root();
         }
-        let mut last_slash = 0;
         let mut idx = 0;
-        for (a, b) in self.0.bytes().zip(other.0.bytes()) {
+        for (a, b) in self.tokens().zip(other.tokens()) {
```

Comment:
I like how simple and neat this turned out!

Reply:
Yea, I was surprised how clean it came out. I definitely borrowed from your initial implementation. 😀